src/Mutator/MutatorFactory.php

```diff
@@ -70,14 +77,36 @@ public function create(array $resolvedMutators): array
                 )
             );
 
-            /** @var string[] $settings */
+            /** @var mixed[] $settings */
             $settings = $config['settings'] ?? [];
             /** @var string[] $ignored */
             $ignored = $config['ignore'] ?? [];
 
-            // TODO: only pass the mutator config if necessary
-            /** @var Mutator $mutator */
-            $mutator = new $mutatorClass(new MutatorConfig($settings));
+            switch ($mutatorClass) {
```

Comment:
I think we can find a more elegant way to do this, but I prefer to keep it simple for this PR